python_modules/dagster/dagster/_core/definitions/antlr_asset_selection/antlr_asset_selection.py

```diff
@@ -113,12 +114,8 @@ def visitFunctionName(self, ctx: AssetSelectionParser.FunctionNameContext):
             return "roots"
 
     def visitKeyExpr(self, ctx: AssetSelectionParser.KeyExprContext):
-        value = self.visit(ctx.value())
-        return AssetSelection.assets(value)
-
-    def visitKeySubstringExpr(self, ctx: AssetSelectionParser.KeySubstringExprContext):
-        value = self.visit(ctx.value())
-        return AssetSelection.key_substring(value)
+        value = self.visit(ctx.keyValue())
+        return KeyWildCardAssetSelection(selected_key_wildcard=value)
```

Comment:
One question for reviewers, should we attach this to the AssetSelection class (eg. `AssetSelection.assetsByWildCardString(value)`)?

Reply:
could potentially include it as a parameter to `AssetSelection.keys(foo, allow_wildcard=True)`

Reply:
Yeah, we should probably do that to avoid exposing `KeyWildCardAssetSelection` and to allow users to access the wildcard functionality programmatically.